doc/source/programs/gdal_raster_reproject.rst

```diff
@@ -129,6 +162,106 @@ Standard options
     Alignment means that xmin / resx, ymin / resy,
     xmax / resx and ymax / resy are integer values.
 
+Advanced options
+++++++++++++++++
+
+.. option:: --srcnodata <SRCNODATA>
+
+    Set nodata masking values for input bands (different values can be supplied
+    for each band). If more than one value is supplied all values should be quoted
+    to keep them together as a single operating system argument.
+    Masked values will not be used in interpolation (details given in :ref:`gdalwarp_nodata`)
+
+    Use a value of ``None`` to ignore intrinsic nodata settings on the source dataset.
+
+    When this option is set to a non-``None`` value, it causes the ``UNIFIED_SRC_NODATA``
+    warping option (see :cpp:member:`GDALWarpOptions::papszWarpOptions`) to be
+    set to ``YES``, if it is not explicitly set.
+
+    If ``--srcnodata`` is not explicitly set, but the source dataset has nodata values,
+    they will be taken into account, with ``UNIFIED_SRC_NODATA`` at ``PARTIAL``
+    by default.
+
+.. option:: --dstnodata <DSTNODATA>
+
+    Set nodata values
+    for output bands (different values can be supplied for each band).  If more
+    than one value is supplied all values should be quoted to keep them together
+    as a single operating system argument.  New files will be initialized to this
+    value and if possible the nodata value will be recorded in the output
+    file. Use a value of ``None`` to ensure that nodata is not defined.
+    If this argument is not used then nodata values will be copied from the source dataset.
+
+.. option:: --addalpha
+
+    Create an output alpha band to identify nodata (unset/transparent) pixels.
+
+.. option:: --wo, --warp-option <NAME>=<VALUE>
+
+    Set a warp option.  The :cpp:member:`GDALWarpOptions::papszWarpOptions` docs show all options.
+    Multiple options may be listed.
+
+.. option:: -to <NAME>=<VALUE>
+
+    Set a transformer option suitable to pass to :cpp:func:`GDALCreateGenImgProjTransformer2`.
+    See :cpp:func:`GDALCreateRPCTransformerV2()` for RPC specific options.
+
+.. option:: --et, --error-threshold <ERROR-THRESHOLD>
+
+    Error threshold for transformation approximation, expressed as a number of
+    source pixels. Defaults to 0.125 pixels unless the ``RPC_DEM`` transformer
+    option is specified, in which case an exact transformer, i.e.
+    ``err_threshold=0``, will be used.
+
+Nodata / source validity mask handling
+--------------------------------------
+
+Invalid values in source pixels, either identified through a nodata value
+metadata set on the source band, a mask band, an alpha band or the use of
+:option:`--srcnodata` will not be used in interpolation.
+The details of how it is taken into account depends on the resampling kernel:
```

Comment:
Maybe mention that in this context alpha is a binary mask and 0 means nodata and all other values mean valid pixels.
I am not sure what is the right thing to do when alpha is used to define partial transparency, should user then use `srcnodata=None`?

Reply:
> Maybe mention that in this context alpha is a binary mask and 0 means nodata and all other values mean valid pixels.

note added

> I am not sure what is the right thing to do when alpha is used to define partial transparency, should user then use srcnodata=None?

Generally you shouldn't mix both nodata and alpha. I don't remember the exact behavior. I prefer not going into that complication in the doc...

Reply:
> Generally you shouldn't mix both nodata and alpha. 

I don't do that, but I have seen some others, maybe experienced PhotoShop users sometimes do. The behavior varies, if I remember right. Agree, don't try to document now.